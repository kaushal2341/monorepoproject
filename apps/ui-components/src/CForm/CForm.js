import React from 'react';
import Form from 'react-bootstrap/Form';
import PropTypes from 'prop-types';

const CForm = props => {
    const {
        _ref,
        as,
        bsPrefix,
        children,
        className,
        id,
        inline,
        noValidate,
        onSubmit,
        validated,
    } = props;
    return (
        <Form
            _ref={_ref}
            as={as}
            bsPrefix={bsPrefix}
            children={children}
            className={className}
            id={"form".concat("_").concat(id)}
            inline={inline}
            noValidate={noValidate}
            onSubmit={onSubmit}
            validated={validated}
        />
    );
};

CForm.defaultProps = {
    onSubmit: () => {
    }
};

CForm.propTypes = {
    _ref: PropTypes.oneOfType([
        PropTypes.func, // for legacy refs
        PropTypes.shape({current: PropTypes.instanceOf(Element)})
    ]),
    as: PropTypes.elementType,
    bsPrefix: PropTypes.string,
    children: PropTypes.node,
    id: PropTypes.string.isRequired,
    inline: PropTypes.bool,
    noValidate: PropTypes.bool,
    onSubmit: PropTypes.func,
    validated: PropTypes.bool,
};

export default CForm;