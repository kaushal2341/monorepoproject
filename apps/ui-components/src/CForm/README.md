# CForm 

Form component made using react-bootstrap's form.

To use it, first

```jsx harmony
 import {CForm} from "@cogent/ui-components";
```
Then simply use it,

```jsx harmony
 let form = (<>
        <input type="text" required/>
        <CButton id="c-button-2" type="submit" name="Submit"/>
   </>);

 <CForm
      id="test"
      children={form}
      onSubmit={()=> {}}
 />
```
You can pass in the form components through children props and submit handler function through onSubmit props.

CForm props:
```
_ref: ReactRef - The Form ref will be forwarded to the underlying element, which means, unless it's rendered as a composite component, it will be a DOM node, when resolved.

as: elementType - You can use a custom element type for this component.,

bsPrefix: string - defaultValue: {'form'} - Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css.,

children: node - Children of Form Component,

className: string - Css classname for additional styling.

id: string - Unique id of the form component,

inline: boolean - defaultValue: false -	Display the series of labels, form controls, and buttons on a single horizontal row,

noValidate: boolean - Determines whether form-data should be validated or not on submission. When set to true,specifies that the form-data should not be validated on submission.
                                                                                                             
onSubmit: function - The method that handles the form submit action.,

validated: boolean - Mark a form as having been validated. Setting it to true will toggle any validation styles on the forms elements.,
```