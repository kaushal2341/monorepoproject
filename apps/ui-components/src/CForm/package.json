{
  "name": "@cogent/c-form",
  "version": "0.0.0",
  "main": "CForm.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
