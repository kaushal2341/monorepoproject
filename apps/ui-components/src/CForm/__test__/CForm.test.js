import CForm from "../CForm";
import checkPropTypes from 'check-prop-types';

expect.addSnapshotSerializer(enzymeSerializer);

describe('CForm Component Tests', () => {

    let wrapper, mockFunction;
    beforeEach(() => {
        mockFunction = jest.fn();
        wrapper = shallow(<CForm id="test" onSubmit={mockFunction}/>);
    });

    test('if CForm Component is defined', () => {
        expect(wrapper.find('CForm')).toBeDefined();
    });

    test('if one Form Component is defined ', () => {
        expect(wrapper.find('#form_test')).toBeDefined();
        expect(wrapper.find('#form_test').length).toBe(1);
    });

    test('if all props required for Form component are defined', () => {
        let propRequired = [
            '_ref',
            'as',
            'bsPrefix',
            'children',
            'className',
            'id',
            'inline',
            'noValidate',
            'onSubmit',
            'validated',
        ];

        let propsOfForm = Object.keys(wrapper.find('#form_test').props());
        propRequired.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfForm[i])
        ));
    });

    test('if onSubmit prop is of type function', () => {
        let result = checkPropTypes(CForm.propTypes, {
            id: "test", onSubmit: () => {
            }
        }, 'prop', CForm.name);
        expect(result).toBeUndefined();
    });

    test('if eventListener function is called on Submit', () => {
        wrapper.find('#form_test').simulate('submit');
        expect(mockFunction).toHaveBeenCalled();
    });

    describe('Snapshot Testing', () => {
        test('if matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
});