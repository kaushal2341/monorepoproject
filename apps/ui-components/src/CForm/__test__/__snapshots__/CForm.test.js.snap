// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CForm Component Tests Snapshot Testing if matches snapshot 1`] = `
<Form
  id="form_test"
  inline={false}
  onSubmit={[MockFunction]}
/>
`;
